File: mtg-card-organizer/src/app/container/models/container.ts
import { Collection } from '../../collection/models/collection';
import { Deck } from '../../deck/models/deck';
import { Permission } from './permission';

export class Container {
  id: number;

  name: string;
  parentId?: number;
  isBookmarked: boolean;
  permission: Permission;

  subContainers: Container[];
  collections: Collection[];
  decks: Deck[];

  public constructor(init?: Partial<Container>) {
    Object.assign(this, init);
  }
}

File: MtgCardOrganizer.Core/Entities/Containers/Container.cs
using System.Collections.Generic;
using MtgCardOrganizer.Core.Entities.Cards;
using MtgCardOrganizer.Core.Entities.Collections;
using MtgCardOrganizer.Core.Entities.Decks;

namespace MtgCardOrganizer.Core.Entities.Containers
{
    public class Container : Entity
    {        
        public string Name { get; set; }
        public bool IsPublic { get; set; }
        public string OwnerUserId { get; set; }

        public int? ParentId { get; set; }
        public Container Parent { get; set; }

        public ICollection<Container> SubContainers { get; set; }
        public ICollection<Collection> Collections { get; set; }
        public ICollection<Deck> Decks { get; set; }

        public ICollection<ContainerUserLink> ContainerUserLinks { get; set; }
    }
}
